diff --git a/examples/sampling_c3/franka_sampling_c3_controller.cc b/examples/sampling_c3/franka_sampling_c3_controller.cc
index d4b5e47..2b67787 100644
--- a/examples/sampling_c3/franka_sampling_c3_controller.cc
+++ b/examples/sampling_c3/franka_sampling_c3_controller.cc
@@ -510,7 +510,7 @@ int DoMain() {
   builder.Connect(controller->get_output_port_dynamically_feasible_curr_plan_object(),
                   dynamically_feasible_curr_plan_object_publisher->get_input_port());
-  builder.Connect(controller->get_output_port_dynamically_feasible_curr_plan_object(),
+  builder.Connect(controller->get_output_port_dynamically_feasible_best_plan_object(),
                   dynamically_feasible_best_plan_object_publisher->get_input_port());
   builder.Connect(target_state_mux->get_output_port(),
                   c3_state_sender->get_input_port_target_state());
